nyxx_interactions/lib/src/builders/CommandPermissionBuilder.dart

```diff
@@ -0,0 +1,26 @@
+part of nyxx_interactions;
+
+/// Used to define permissions for a particular command.
+class CommandPermissionBuilder extends Builder {
+  /// The role ID to change the permissions for
+  Snowflake? role;
+
+  /// The user'd ID to change the permissions for
+  Snowflake? user;
+
+  /// Does the role have permission to use the command
+  final bool hasPermission;
+
+  /// A permission for a single user that can be used in [SlashCommandBuilder]
+  CommandPermissionBuilder.forUser(this.user, {this.hasPermission = true});
+
+  /// A permission for a single role that can be used in [SlashCommandBuilder]
+  CommandPermissionBuilder.forRole(this.role, {this.hasPermission = true});
```

Comment:
it would be better to have abstrat builder that would wrap that two cases because its kind of cofusing to have both user and role in same class. 

ICommandPermissionBuilder
 * RoleCommandPermissionBuilder
 *  UserCommandPermissionBuilder
 
 Then ICommandPermissionBuilder could have getter for id, type and permission and factory custructor for both subclasses

Reply:
Done initial implementation of this.